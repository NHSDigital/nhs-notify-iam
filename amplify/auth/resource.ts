import { DeepPartialAmplifyGeneratedConfigs } from '@aws-amplify/plugin-types';
import { ClientConfig } from '@aws-amplify/client-config';
import { defineAuth } from '@aws-amplify/backend';

export const remoteAuthConfig: DeepPartialAmplifyGeneratedConfigs<ClientConfig> =
  {
    auth: {
      aws_region: 'eu-west-2',
      user_pool_id: process.env.USER_POOL_ID,
      user_pool_client_id: process.env.USER_POOL_CLIENT_ID,
    },
  };

export const sandboxAuthConfig = {
  auth: defineAuth({
    loginWith: {
      email: true,
    },
  }),
};
