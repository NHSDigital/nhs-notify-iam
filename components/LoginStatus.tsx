'use client';

import { Header } from 'nhsuk-react-components';
import React from 'react';
import { useAuthenticator } from '@aws-amplify/ui-react';

export default function LoginStatus() {
  const { authStatus } = useAuthenticator();
  switch (authStatus) {
    case 'authenticated':
      return <Header.NavItem href="/auth/signout">
        Sign out
      </Header.NavItem>;
    case 'unauthenticated':
      return <Header.NavItem href={`/auth/?redirect=${location.pathname}`}>
        Sign in
      </Header.NavItem>;
    default:
      return null;
  }
}
