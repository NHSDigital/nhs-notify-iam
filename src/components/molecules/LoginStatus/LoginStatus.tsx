'use client';

import React from 'react';
import { useAuthenticator } from '@aws-amplify/ui-react';
import { getBasePath } from '@/utils/get-base-path';

export const LoginStatus = () => {
  const { authStatus } = useAuthenticator();

  if (authStatus === 'authenticated') {
    return (
      <a
        className='nhsuk-account__login--link'
        href={`${getBasePath()}/signout`}
      >
        Log out
      </a>
    );
  }

  return (
    <a className='nhsuk-account__login--link' href={`${getBasePath()}`}>
      Log in
    </a>
  );
};
