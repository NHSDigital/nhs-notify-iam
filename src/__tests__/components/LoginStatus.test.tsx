import { mockDeep } from 'jest-mock-extended';
import { UseAuthenticator, useAuthenticator } from '@aws-amplify/ui-react';
import { render, screen } from '@testing-library/react';
import { AuthSession } from '@aws-amplify/auth';
import { LoginStatus } from '../../components/molecules/LoginStatus/LoginStatus';

jest.mock('@aws-amplify/ui-react');
jest.mock('aws-amplify/auth', () => ({
  fetchAuthSession: async () =>
    mockDeep<AuthSession>({
      tokens: {
        idToken: {
          payload: {
            email: 'test-email',
          },
        },
      },
    }),
}));

const useAuthenticatorMock = jest.mocked(useAuthenticator);

test('LoginStatus - authenticated', async () => {
  useAuthenticatorMock.mockReturnValue(
    mockDeep<UseAuthenticator>({
      authStatus: 'authenticated',
    })
  );

  const container = render(<LoginStatus />);

  await screen.findByText('Log out');

  expect(container.asFragment()).toMatchSnapshot();
});

test('LoginStatus - unauthenticated', async () => {
  useAuthenticatorMock.mockReturnValue(
    mockDeep<UseAuthenticator>({
      authStatus: 'unauthenticated',
    })
  );

  const container = render(<LoginStatus />);

  await screen.findByText('Log in');

  expect(container.asFragment()).toMatchSnapshot();
});

test('LoginStatus - configuring', async () => {
  jest.mocked(useAuthenticator).mockReturnValue(
    mockDeep<UseAuthenticator>({
      authStatus: 'configuring',
    })
  );

  const container = render(<LoginStatus />);

  await screen.findByText('Log in');

  expect(container.asFragment()).toMatchSnapshot();
});
