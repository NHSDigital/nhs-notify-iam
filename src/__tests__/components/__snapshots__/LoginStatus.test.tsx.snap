// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginStatus - authenticated 1`] = `
<DocumentFragment>
  <a
    class="nhsuk-account__login--link"
    href="/auth/signout"
  >
    Log out
  </a>
</DocumentFragment>
`;

exports[`LoginStatus - configuring 1`] = `
<DocumentFragment>
  <a
    class="nhsuk-account__login--link"
    href="/auth"
  >
    Log in
  </a>
</DocumentFragment>
`;

exports[`LoginStatus - unauthenticated 1`] = `
<DocumentFragment>
  <a
    class="nhsuk-account__login--link"
    href="/auth"
  >
    Log in
  </a>
</DocumentFragment>
`;
